### ISSUE TITLE
Wrong sorting in data viewer if columns in view don't start from the first

### ISSUE THREAD

**Issue by stefanocoretta** [rstudio/rstudio#4682]
### System details

    RStudio Edition : Desktop
    RStudio Version : 1.2.1502
    OS Version      : macOS 10.14.4 (18E226)
    R Version       : 3.5.2 (2018-12-20)

### Steps to reproduce the problem

1. Read a dataframe with more than 50 columns.
2. Visualise a range of columns that does not include the first colum.
2. Sort in the data viewer any numeric column.

### Describe the problem in detail

When trying to sort a column in the data viewer, and the columns in view do not start with the first column, the sorting is wrong.

<img width="1312" alt="example" src="https://user-images.githubusercontent.com/6429302/56505701-a1d39b00-651c-11e9-9cd2-11e2273e373e.png">

### Describe the behavior you expected

The expected behaviour should be correct sorting of the numeric values in the columns.

```
─ Session info ───────────────────────────────────────────────────────
 setting  value                       
 version  R version 3.5.2 (2018-12-20)
 os       macOS Mojave 10.14.4        
 system   x86_64, darwin15.6.0        
 ui       RStudio                     
 language (EN)                        
 collate  en_GB.UTF-8                 
 ctype    en_GB.UTF-8                 
 tz       Europe/Rome                 
 date     2019-04-22                  

─ Packages ───────────────────────────────────────────────────────────
 package           * version    date       lib source        
 assertthat          0.2.1      2019-03-21 [1] CRAN (R 3.5.2)
 backports           1.1.4      2019-04-10 [1] CRAN (R 3.5.2)
 callr               3.2.0      2019-03-15 [1] CRAN (R 3.5.2)
 cli                 1.1.0      2019-03-19 [1] CRAN (R 3.5.2)
 coretta2018itapol * 0.0.0.9000 2019-04-14 [1] local         
 crayon              1.3.4      2017-09-16 [1] CRAN (R 3.5.0)
 desc                1.2.0      2018-05-01 [1] CRAN (R 3.5.0)
 devtools            2.0.2      2019-04-08 [1] CRAN (R 3.5.2)
 digest              0.6.18     2018-10-10 [1] CRAN (R 3.5.0)
 fs                  1.2.7      2019-03-19 [1] CRAN (R 3.5.2)
 glue                1.3.1      2019-03-12 [1] CRAN (R 3.5.2)
 magrittr            1.5        2014-11-22 [1] CRAN (R 3.5.0)
 memoise             1.1.0      2017-04-21 [1] CRAN (R 3.5.0)
 packrat             0.5.0      2018-11-14 [1] CRAN (R 3.5.0)
 pkgbuild            1.0.3      2019-03-20 [1] CRAN (R 3.5.2)
 pkgload             1.0.2      2018-10-29 [1] CRAN (R 3.5.0)
 prettyunits         1.0.2      2015-07-13 [1] CRAN (R 3.5.0)
 processx            3.3.0      2019-03-10 [1] CRAN (R 3.5.2)
 ps                  1.3.0      2018-12-21 [1] CRAN (R 3.5.0)
 R6                  2.4.0      2019-02-14 [1] CRAN (R 3.5.2)
 Rcpp                1.0.1      2019-03-17 [1] CRAN (R 3.5.2)
 remotes             2.0.4      2019-04-10 [1] CRAN (R 3.5.2)
 rlang               0.3.4      2019-04-07 [1] CRAN (R 3.5.2)
 rprojroot           1.3-2      2018-01-03 [1] CRAN (R 3.5.0)
 rstudioapi          0.10       2019-03-19 [1] CRAN (R 3.5.2)
 sessioninfo         1.1.1      2018-11-05 [1] CRAN (R 3.5.0)
 testthat            2.0.1      2018-10-13 [1] CRAN (R 3.5.0)
 usethis             1.5.0      2019-04-07 [1] CRAN (R 3.5.2)
 withr               2.1.2      2018-03-15 [1] CRAN (R 3.5.0)

```

**Comment by willtudorevans** [comment]
Filtering doesn't seem to work either.

**Comment by SimonDedman** [comment]
The problem is that it's using relative referencing of column numbers from the end, but the end is hard coded to be <=50 so if you sort on:
column 50 of 50 ("last" = -0) it works
column 51 of 51, -0 should be 51 but it's 50.
I'm trying to sort on column 53 of 56, -3, sorting on 47.

**Comment by JLewyckyj** [comment]
It seems kind of ridiculous that this doesn't work. What were you fixing by creating all this new functionality, that just so happens to ruin basic functionality that previously worked? Do you have any sense?

**Comment by rich-rstudio** [comment]
verified on Version 1.3.921

**Comment by m-macaskill** [comment]
This bug is still present in RStudio version 1.4.1717.

**Comment by wizbionet** [comment]
I just noticed this bug in  RStudio 2023.06.1+524 "Mountain Hydrangea" Release (547dcf861cac0253a8abb52c135e44e02ba407a1, 2023-07-07) for windows

**Comment by ronblum** [comment]
@willtudorevans This bug is covered by issue #13328 and is included as backport issue #13345 for Mountain Hydrangea. The fix will be available in the second Mountain Hydrangea patch, 2023.06.2, which we're working on right now—it will be released soon. If you'd like to try it out, [daily builds are available for it](https://dailies.rstudio.com/rstudio/mountain-hydrangea/), though keep in mind that they're for testing purposes only.
